File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        counter = 0
        if needle not in haystack:
            return -1
        else:
            for i in haystack:
                if needle == haystack[counter:len(needle)+counter]:
                    return counter
                else:
                    counter += 1
        
    

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if(needle.length > haystack.length) return -1
    for(let i=0; i <= haystack.length - needle.length; i++){
        if(haystack.slice(i, needle.length + i) == needle){
            return i
        }
    }
    return -1
};